### PR TITLE
Added a task to connect a repository to an ArgoCD server

### PR DESCRIPTION
**Pull request by mestredelpino**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by mestredelpino** [comment]
Hi, I am not sure why it is returning that error that expects the task to be under "0.2". If this is the first version, shouldn't 0.1 be ok?

File: task/argocd-task-connect-repo/0.1/argocd-task-connect-repo.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: argocd-task-connect-repo
+  namespace: argocd
+  labels:
+    app.kubernetes.io/version: "0.2"
```
Comment (vinamra28) [inline task/argocd-task-connect-repo/0.1/argocd-task-connect-repo.yaml:7]:
```suggestion
    app.kubernetes.io/version: "0.1"
```

File: task/argocd-task-connect-repo/0.1/argocd-task-connect-repo.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Deployment
+    tekton.dev/tags: deploy
+    tekton.dev/displayName: "argocd"
```
Comment (vinamra28) [inline task/argocd-task-connect-repo/0.1/argocd-task-connect-repo.yaml:12]: not important but should we change this displayName to something else and meaningful?

Comment (mestredelpino) [reply task/argocd-task-connect-repo/0.1/argocd-task-connect-repo.yaml:12]: Yes, sorry, good point. perhaps argocd-repository?

**Pull request closed without merging**